### PR TITLE
Add ADR on TLS authentication

### PR DESCRIPTION
**Pull request by sbernauer**
For https://github.com/stackabletech/zookeeper-operator/issues/466

### CODE REVIEW

File: modules/contributor/pages/adr/ADR016-tls-authentication.adoc [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        replicas: 1
+        config:
+          myidOffset: 20
+  version: 3.8.0
+  stopped: false
+---
+apiVersion: authentication.stackable.tech/v1alpha1
+kind: AuthenticationClass
+metadata:
+  name: zookeeper-tls-mutual
```
Comment (maltesander) [inline modules/contributor/pages/adr/ADR016-tls-authentication.adoc:235]:
```suggestion
  name: zookeeper-tls
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Maybe for completion we can add the referred `tls` SecretClass:
```
apiVersion: secrets.stackable.tech/v1alpha1
kind: SecretClass
metadata:
  name: tls
spec:
  backend:
    autoTls:
      ca:
        secret:
          name: secret-provisioner-tls-ca
          namespace: default
        autoGenerate: true
```

And there are some reference errors (zookeeper-tls <--> zookeeper-tls-mutual) in the image that may confuse people.

**Comment by sbernauer** [comment]
Thanks for the Review, should be addressed

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request merged**